WordPress/Classes/ViewRelated/Blog/Blog List/BlogListViewController.h

```diff
@@ -2,9 +2,12 @@
 
 @class Blog;
 
+@protocol ScenePresenter;
+
 @interface BlogListViewController : UIViewController
 
 @property (nonatomic, strong) Blog *selectedBlog;
+@property (nonatomic, strong) id<ScenePresenter> scenePresenter;
```

Comment:
This name seems kind of generic. It seems like there are a lot of things that are presented by `BlogList`, `WPTabBarController` and `BlogDetails`. Would it make more sense to specify that this as the "Me" presenter, or would the scene presenter be more generic for other presentations?

Reply:
Thanks for bringing this up @bjtitus ! It's a very good point.

The type is generic as it was intended to be, but there's a bit of misuse.
- The name of the property is generic, but in reality it's used specifically for the `Me` button
- Plus, there's a bit of duplication and injection is not properly used. A few things can be done to improve it:
  - leave the type generic but name the parameters meScenePresenter as you say (and do this in all the three cases: BlogList, BlogDetails, TabBarController)

  - In Addition:
    - Add a new initializer to BlogListViewController and inject the dependency (as I already did for BlogDetailsViewController)
    - Because the `ScenePresenter` instance is always a `MeScenePresenter` instance for all the three cases (BlogList, BlogDetail and TabBarController), and there would not be more than one presentation at a time, then we can use `WPTabBarController` at the "outer scope" and inject the same instance from there into BlogList and from BlogList into BlogDetails. It's not super ideal cause WPTabBarController is a singleton, but it's an alternative of having an instance of `MeScenePresenter` in each one of them. What do you think?

Reply:
It's really the just the name I'm wondering about. Since the property is only called from the locations presenting the Me screen, it seems like it could be more specific.

As far as the injection stuff goes:

> Add a new initializer to BlogListViewController and inject the dependency

I think this might make sense as a follow up to clean this controller up. It might be better as a separate issue, though, so it isn't intertwined with the iA work.

> we can use WPTabBarController as the "outer scope" and inject the same instance from there into BlogList and from BlogList into BlogDetails

This could make sense. Any issue to remove the WPTabBarController singleton would probably be a good place to do that.